core/src/main/scala/kafka/log/remote/RemoteIndexCache.scala

```diff
@@ -136,12 +176,14 @@ class RemoteIndexCache(maxSize: Int = 1024, remoteStorageManager: RemoteStorageM
       val offset = name.substring(0, firstIndex).toInt
       val uuid = Uuid.fromString(name.substring(firstIndex + 1, name.lastIndexOf('_')))
 
-      if(!entries.containsKey(uuid)) {
+      if (internalCache.getIfPresent(uuid) == null) {
```

Comment:
did you mean `internalCache.asMap().containsKey(uuid)`? That would be an existence check, as this is an entry read so it results in a policy hit/miss behavior.

Reply:
Good point. Updated.